packages/perspective/src/transaction/classify.ts

```diff
@@ -1,92 +1,196 @@
-import { TransactionView } from '@penumbra-zone/protobuf/penumbra/core/transaction/v1/transaction_pb';
+import {
+  TransactionView,
+  ActionView,
+} from '@penumbra-zone/protobuf/penumbra/core/transaction/v1/transaction_pb';
+import { MsgRecvPacket } from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
 import { TransactionClassification } from './classification.js';
 
-export const classifyTransaction = (txv?: TransactionView): TransactionClassification => {
+export interface ClassificationReturn {
+  /** A type of the main action that defines a transaction */
+  type: TransactionClassification;
+  /** A main action that defines the transaction */
+  action?: ActionView;
+}
+
+/**
+ * Takes a transaction view, finds the most relevant action that defines its type, and returns it.
+ */
+export const classifyTransaction = (txv?: TransactionView): ClassificationReturn => {
```

Comment:
a powerful function, it was returning only a classified type of a transaction. I added the return of the most relevant action type given transaction